### PR TITLE
Add linter check/support for SPDX headers (#3028)

### PR DESCRIPTION
**Pull request by maronasn**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: tools/roxvet/analyzers/verifylicenseidentifier/analyzer.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+var (
+	copyrightHeaderRegex := regexp.MustCompile('^//+(\s|)(?:Copyright)')
+	licenseIDHeaderRegex := regexp.MustCompile('^//+(\s|)(?:SPDX-License-Identifier')
+)
+
+const doc = 'verifies if source file has appropriae SPDX header'
```
Comment (porridge) [inline tools/roxvet/analyzers/verifylicenseidentifier/analyzer.go:17]:
```suggestion
const doc = 'verifies if source file has appropriate copyright and SPDX license headers'
```

**Review by porridge — COMMENTED** [review]
@maronasn can you please elaborate on the goal of this change in the description?
Also, perhaps I'm missing something but it does not look like most (any?) of stackrox source code adheres to the rules you're adding?

**Comment by Maddosaurus** [comment]
Ah! This seems to be the PR for https://github.com/stackrox/stackrox/issues/3028. 
So the goal is to add linter support for when we roll out SPDX headers to our codebase.

**Comment by Maddosaurus** [comment]
Please have a look at the failing style checks @maronasn - these should be rather quick fixes 😄

**Comment by porridge** [comment]
@maronasn do you still intend to work on this change?

**Comment by Maddosaurus** [comment]
As this PR has been inactive for a longer time now, I'll be closing it. Should you be interested to continue work on this, please feel free to reopen this PR or create a new one :smile:

**Pull request closed without merging**